api/organisations/tests/test_views.py

```diff
@@ -952,6 +960,95 @@ def test_organisation_get_influx_data(
     assert response.json() == {"events_list": []}
 
 
+@pytest.mark.parametrize(
+    "plan_id, max_seats, max_api_calls, max_projects, is_updated",
+    [
+        ("plan-id", 3, 100, 3, False),
+        ("updated-plan-id", 5, 500, 10, True),
+    ],
+)
+@mock.patch("organisations.models.get_plan_meta_data")
+@mock.patch("organisations.views.extract_subscription_metadata")
+def test_when_plan_is_changed_max_seats_and_max_api_calls_are_updated(
+    mock_get_subscription_metadata,
+    mock_get_plan_meta_data,
+    subscription,
+    admin_client,
+    organisation,
+    plan_id,
+    max_seats,
+    max_api_calls,
+    max_projects,
+    is_updated,
+):
+    # Given
+    chargebee_email = "chargebee@test.com"
+    url = reverse("api-v1:chargebee-webhook")
+    updated_at = datetime.now(tz=UTC) - timedelta(
+        days=1
+    )  # The timestamp representing the last update time, one day ago from the current time.
+
+    mock_get_plan_meta_data.return_value = {
+        "seats": max_seats,
+        "api_calls": max_api_calls,
+    }
+    mock_get_subscription_metadata.return_value = ChargebeeObjMetadata(
+        seats=max_seats,
+        api_calls=max_api_calls,
+        projects=max_projects,
+        chargebee_email=chargebee_email,
+    )
+
+    data = {
+        "content": {
+            "subscription": {
+                "status": "active",
+                "id": subscription.subscription_id,
+                "plan_id": plan_id,
+            },
+            "customer": {"email": chargebee_email},
+        }
+    }
+
+    if is_updated:
+        subscription_information_cache = (
+            OrganisationSubscriptionInformationCache.objects.create(
+                organisation=organisation,
+                allowed_seats=1,
+                allowed_30d_api_calls=10,
+                allowed_projects=1,
+                chargebee_email=chargebee_email,
+                chargebee_updated_at=updated_at,
+                influx_updated_at=None,
+            )
+        )
```

Comment:
In general, I don't love controlling test flow like this, based on parameters in the test, but I think it makes sense here to reduce code duplication.